File: A3/10.c
#include <stdio.h>

int main(){

    int x[3] = {0};
    int count = 0;
    int position = 1;
    scanf("%d %d %d",&x[0],&x[1],&x[2]);

    do{
        if(position + x[1] <= x[0]){
            position += x[1];
        }
        else{
            position = (position + x[1]) - x[0];
        }
        count++;

    }while(position != 1 && position != x[2]);

    if(position == x[2]){count++;}

    printf("%d\n",count);
    return 0;
}

File: A3/10.py

inputs = input().split()
count = 0
position = 0
while position != 0 and position != inputs[2]:
    if position + inputs[1] <= inputs[0]:
        position += inputs[1]
    else:
        position = (position + inputs[1]) - inputs[0]
    count += 1

print(count)
        
